app/models/user.rb

```diff
@@ -76,6 +76,14 @@ def current_organisation_mou_signature
     mou_signatures.find_by(organisation:)
   end
 
+  def role_changed_to_editor?
+    role_changed_to_editor = editor? && !versions.empty? && versions.last.reify.role != "editor"
```

Comment:
This uses the PaperTrail version history to determine if the role has been changed to editor since the last version was saved. It then saves a version with no changes so that the next time it is checked `role_changed_to_editor` is false. 

This seems a bit hacky, but the alternatives either didn't work the way we'd want (e.g. flash messages would appear on each login, session storage would lead to users missing the message if their session expired before they logged in after upgrade), or involved persisting something specific for this feature that describes state change rather than state (e.g. `role_changed` rather than `role`) which seems worse to me. 

I'd welcome any suggestions!

Reply:
Using PaperTrail doesn't sound like too bad an option to me, although I'd be interested to know what @aliuk2012 thinks. 

The only thing I would suggest is using a custom event, so it's maybe more clear in the database what's happened?

https://github.com/paper-trail-gem/paper_trail#the-versionsevent-column

Reply:
Good suggestion! I'd like that, but on investigation it looks like `paper_trail.save_with_version` always saves with an update event: https://www.rubydoc.info/gems/paper_trail/11.0.0/PaperTrail%2FRecordTrail:save_with_version

Reply:
Hmm, looking at the source code it appears to still be overridable?

`save_with_version` does (indirectly) create a `Events::Update` instance [[1](https://github.com/paper-trail-gem/paper_trail/blob/47dbc2271c7055fde6f0fe7d3c8bf44f62cbef63/lib/paper_trail/record_trail.rb#L152), [2](https://github.com/paper-trail-gem/paper_trail/blob/47dbc2271c7055fde6f0fe7d3c8bf44f62cbef63/lib/paper_trail/record_trail.rb#L101), [3](https://github.com/paper-trail-gem/paper_trail/blob/47dbc2271c7055fde6f0fe7d3c8bf44f62cbef63/lib/paper_trail/record_trail.rb#L218)], but the actual value of the `event` column is still customisable:

```
          event: @record.paper_trail_event || "update",
```

https://github.com/paper-trail-gem/paper_trail/blob/47dbc2271c7055fde6f0fe7d3c8bf44f62cbef63/lib/paper_trail/events/update.rb#L29

Reply:
Hmm interesting. I tried it with no luck. I must've done something silly, I'll look again.

Reply:
Got it working, thanks for the suggestion!